paddle/fluid/imperative/gradient_accumulator.cc

```diff
@@ -65,6 +67,177 @@ class TensorAddFunctor : public boost::static_visitor<> {
   T* y_;
 };
 
+void SelectedRowsAddToTensor(const framework::Variable& src,
+                             framework::Variable* dst) {
+  auto* dst_tensor = dst->GetMutable<framework::LoDTensor>();
+  auto& src_selectedrows = src.Get<framework::SelectedRows>();
+  auto place = dst_tensor->place();
+  auto data_type = src_selectedrows.value().type();
+  platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
+
+  if (paddle::platform::is_gpu_place(place)) {
+#ifdef PADDLE_WITH_CUDA
+#define PADDLE_SelectedRowsAddToTensor_MACRO(cpp_type)                      \
```

Comment:
why not use template

Reply:
if you want use MACRO, I think it's better to extract it and write it outside the function is easier to read

Reply:
I see, but I think the code can be simplified

Reply:
I see, but the code can be simplified. https://github.com/PaddlePaddle/Paddle/pull/21078/commits/b8aa72c56a9f71388ad150f3e544326fd26ccda5

Reply:
done, thx